tests/aws/test_error_injection.py

```diff
@@ -35,78 +35,79 @@ def test_kinesis_error_injection(
-        finally:
-            table.delete()
+    def test_dynamodb_read_error_injection(self, monkeypatch, aws_client, dynamodb_create_table):
+        table_name = dynamodb_create_table()["TableDescription"]["TableName"]
+
+        partition_key = short_uid()
+        self.assert_zero_probability_read_error_injection(
+            aws_client.dynamodb, table_name, partition_key
+        )
+
+        # with a probability of 1, always throw errors
+        monkeypatch.setattr(config, "DYNAMODB_READ_ERROR_PROBABILITY", 1.0)
+        with pytest.raises(ClientError) as exc:
+            aws_client.dynamodb.get_item(
+                TableName=table_name, Key={PARTITION_KEY: {"S": partition_key}}
+            )
+        exc.match("ProvisionedThroughputExceededException")
 
     @markers.aws.only_localstack
-    def test_dynamodb_write_error_injection(self, monkeypatch):
-        table = self.get_dynamodb_table()
+    def test_dynamodb_write_error_injection(self, monkeypatch, aws_client, dynamodb_create_table):
+        table_name = dynamodb_create_table()["TableDescription"]["TableName"]
 
-        try:
-            # by default, no errors
-            test_no_errors = table.put_item(Item={PARTITION_KEY: short_uid(), "data": "foobar123"})
-            assert test_no_errors["ResponseMetadata"]["HTTPStatusCode"] == 200
-
-            # with a probability of 1, always throw errors
-            monkeypatch.setattr(config, "DYNAMODB_WRITE_ERROR_PROBABILITY", 1.0)
-            with pytest.raises(ClientError) as exc:
-                table.put_item(Item={PARTITION_KEY: short_uid(), "data": "foobar123"})
-            exc.match("ProvisionedThroughputExceededException")
-
-            # BatchWriteItem throws ProvisionedThroughputExceededException if ALL items in Batch are Throttled
-            with pytest.raises(ClientError) as exc:
-                with table.batch_writer() as batch:
-                    for _ in range(3):
-                        batch.put_item(
-                            Item={
-                                PARTITION_KEY: short_uid(),
-                                "data": "foobar123",
-                            }
-                        )
-            exc.match("ProvisionedThroughputExceededException")
-        finally:
-            table.delete()
+        # by default, no errors
+        test_no_errors = aws_client.dynamodb.put_item(
+            TableName=table_name,
+            Item={PARTITION_KEY: {"S": short_uid()}, "data": {"S": "foobar123"}},
+        )
+        assert test_no_errors["ResponseMetadata"]["HTTPStatusCode"] == 200
 
-    def get_dynamodb_table(self):
-        # set max_attempts=1 to speed up the test execution
-        dynamodb = aws_stack.connect_to_resource("dynamodb", config=self.retry_config())
-        table_name = f"table-{short_uid()}"
-        resources.create_dynamodb_table(table_name, partition_key=PARTITION_KEY)
-        return dynamodb.Table(table_name)
```

Comment:
This helper has been removed altogether because it used the `connect_to_resource` utility. Instead, the tests now rely on the already existing `dynamodb_create_table` fixture.